extensions/wsc/src/main/java/org/mule/extension/ws/internal/connection/ClientFactory.java

```diff
@@ -4,100 +4,62 @@
 import org.mule.extension.ws.internal.interceptor.OutputSoapHeadersInterceptor;
 import org.mule.extension.ws.internal.interceptor.SoapActionInterceptor;
 import org.mule.extension.ws.internal.interceptor.StreamClosingInterceptor;
-import org.mule.extension.ws.internal.introspection.WsdlIntrospecter;
 import org.mule.extension.ws.internal.transport.WscTransportFactory;
 import org.mule.runtime.api.connection.ConnectionException;
-import org.mule.runtime.api.connection.ConnectionValidationResult;
 
 import java.util.List;
-import java.util.Map;
 
 import javax.wsdl.Port;
 import javax.wsdl.extensions.http.HTTPAddress;
 import javax.wsdl.extensions.soap.SOAPAddress;
 import javax.wsdl.extensions.soap12.SOAP12Address;
-import javax.xml.namespace.QName;
 
 import org.apache.cxf.binding.Binding;
 import org.apache.cxf.binding.soap.interceptor.CheckFaultInterceptor;
 import org.apache.cxf.binding.soap.interceptor.Soap11FaultInInterceptor;
 import org.apache.cxf.binding.soap.interceptor.Soap12FaultInInterceptor;
 import org.apache.cxf.endpoint.Client;
-import org.apache.cxf.endpoint.Endpoint;
 import org.apache.cxf.interceptor.Interceptor;
 import org.apache.cxf.interceptor.WrappedOutInterceptor;
-import org.apache.cxf.message.Exchange;
 import org.apache.cxf.message.Message;
 import org.apache.cxf.phase.PhaseInterceptor;
-import org.apache.cxf.service.model.BindingOperationInfo;
 
 /**
- * A connection with a web service for consuming it's exposed resources.
+ * Factory class that creates instances of cxf {@link Client}s.
  *
  * @since 4.0
  */
-public class WscConnection {
-
-  private final Client client;
-  private final String wsdlLocation;
-  private final SoapVersion soapVersion;
-  private final WsdlIntrospecter wsdlIntrospecter;
-
-  public WscConnection(String wsdlLocation,
-                       String address,
-                       String serviceName,
-                       String portName,
-                       SoapVersion soapVersion,
-                       boolean mtomEnabled)
-      throws ConnectionException {
-    this.wsdlIntrospecter = new WsdlIntrospecter(wsdlLocation, serviceName, portName);
-    this.wsdlLocation = wsdlLocation;
-    this.soapVersion = soapVersion;
+final class ClientFactory {
+
+  /**
+   * Creates a new instance of a cxf {@link Client} for the given address ans soap version.
+   * <p>
+   * Adds all the custom {@link Interceptor}s to work with the {@link WebServiceConsumer}.
+   *
+   * @param address     the address of the web service
+   * @param port        the configured port from the web service definition
+   * @param soapVersion the soap version of the web service
+   * @param mtomEnabled if should enable MTOM attachments or not.
+   * @return a new configured {@link Client}.
+   * @throws ConnectionException if the client couldn't be created.
+   */
+  static Client create(String address, Port port, SoapVersion soapVersion, boolean mtomEnabled) throws ConnectionException {
```

Comment:
it actually was an instance method and suggested to make this an static method since the factory had no state and having to create a new instance just for this seemed unnecessary